lib/puppet/type/file/content.rb

```diff
@@ -119,6 +120,11 @@ def insync?(is)
       result
     end
 
+    def property_matches?(current, desired)
+      return super unless [:mtime, :ctime].include?(checksum_type)
+      DateTime.parse(current) >= DateTime.parse(desired)
```

Comment:
I'll look at this more. I haven't checked whether Windows and Linux mtimes match.

When they're not convertible to a time an exception is thrown, so I should probably catch that and have a fallback super comparison. The exception gets caught higher up, but will always assume the match failed.